### PR TITLE
added various features

### PR DESCRIPTION
**Pull request by narharikale**

### CODE REVIEW

File: src/frontend/components/editModal/EditModal.js [hunk 1]
```diff
@@ -26,20 +26,21 @@ import { updateUser } from '../../features/user/userSlice';
 
 function EditModal( { isOpen , onClose , singleUser } ) {
 
+  console.log(singleUser , " single")
```
Comment (horsemaker) [inline src/frontend/components/editModal/EditModal.js]: Remove this console if not needed.

File: src/frontend/features/bookmarks/bookmarksSlice.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        [deleteFromBookmark.rejected] : ( state , { error } ) =>{
+            state.status = "rejected";
+            state.error = error.message ;
+        },
+        [getBookmarks.pending] : (state) => {
+            state.status = 'loading';
+        },
+        [getBookmarks.fulfilled] : ( state , { payload } ) =>{
+            state.bookmarks = payload.bookmarks;
+			state.status = "success";
```
Comment (horsemaker) [inline src/frontend/features/bookmarks/bookmarksSlice.js]: The indentation here doesn't seem to be right.

**Review by horsemaker — APPROVED** [review]
The UI looks fantastic. 🔥

File: src/frontend/components/editModal/EditModal.js (hunk 1 above)
Comment (narharikale) [reply src/frontend/components/editModal/EditModal.js]: removed

File: src/frontend/features/bookmarks/bookmarksSlice.js (hunk 2 above)
Comment (narharikale) [reply src/frontend/features/bookmarks/bookmarksSlice.js]: fixed!

**Review by rahulmistry751 — APPROVED** [review]
Code looks clean and well structured

**Pull request merged**